File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab7/Task2/Java/MatrixThread.java
package lab7.threading;

public class MatrixThread implements Runnable {
    private Thread thread;
    private int[][] matrix;
    private int row, delay, sum;
    
    public MatrixThread(
        int[][] matr, 
        int startRow,
        String name,
        int priority,
        int delay) {
        
        matrix = matr;
        row = startRow;
        this.delay = delay;
        sum = 0;
        
        thread = new Thread(this);
        thread.setName(name);
        thread.setPriority(priority);
    }
    
    public Thread getThread() {
        return thread;
    }
    
    public int getSum() {
        return sum;
    }
    
    public void run() {
        for (int i = row; i < matrix.length; i += 2) {
            for (int j = 0; j < matrix[i].length; j++) {
                sum += matrix[i][j];
                
                try {
                    Thread.sleep(delay);
                }
                catch (InterruptedException e) {
                    System.out.println(e.getMessage());
                }
            }
        }
        
        System.out.println(Thread.currentThread() + ", Sum: " + sum);
    }
}

File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab7/Task2/C#/MatrixThread.cs
using System;
using System.Threading;

namespace CSharpThreading {
    class MatrixThread {
        public Thread thread;
        private int[,] matrix;
        private int delay, row, sum;

        public Thread GetThread { get { return thread; } }
        public int Sum { get { return sum; } }

        public MatrixThread(
            int[,] matr,
            int startRow,
            string name,
            ThreadPriority priority,
            int delay) {

            sum = 0;
            matrix = matr;
            row = startRow;
            this.delay = delay;

            thread = new Thread(this.Run);
            thread.Name = name;
            thread.Priority = priority;
        }

        void Run() {
            for (int i = row; i < matrix.GetLength(0); i += 2)
                for (int j = 0; j < matrix.GetLength(1); j++) {
                    sum += matrix[i, j];

                    Thread.Sleep(delay);
                }

            Console.WriteLine("Thread[{0}, {1}], Sum: {2}",
                Thread.CurrentThread.Name,
                Thread.CurrentThread.Priority, sum);
        }
    }
}
